paasta_tools/kubernetes_tools.py

```diff
@@ -3432,3 +3433,25 @@ def update_crds(
             success = False
 
     return success
+
+
+def get_kubernetes_secret_name(
+    service_name: str, secret_name: str, namespace: str = "paasta"
+) -> str:
+    service = sanitise_kubernetes_name(service_name)
+    sanitised_secret = sanitise_kubernetes_name(secret_name)
+    name = f"{namespace}-secret-{service}-{sanitised_secret}"
+    return name
+
+
+def get_kubernetes_secret(secret_name: str, service_name: str, cluster: str) -> str:
+    os.environ["KUBECONFIG"] = DEFAULT_KUBECTL_CONFIG_PATH
+    os.environ["KUBECONTEXT"] = cluster
```

Comment:
we probably want to do the kubeconfig/kubecontext stuff in cli/cmds/secret.py - otherwise, anyone trying to use this in paasta internals is going to potentially overwrite environment variables in undesirable ways

e.g., if someone where to call this in setup_kubernetes_job, our kube masters would switch from using the paasta admin creds to using the user creds here (and also need to start passing around cluster arguments)

Reply:
To double check I understand the suggestion, modifying the KubeClient constructor to this:
`class KubeClient:
    def __init__(self, component: Optional[str] = None, config_file: Optional[str] = None, context: Optional[str] = None) -> None:
        if config_file and context:
            kube_config.load_kube_config(
                config_file=config_file,
                context=context
            )
        else:
            kube_config.load_kube_config(
                config_file=os.environ.get("KUBECONFIG", KUBE_CONFIG_PATH),
                context=os.environ.get("KUBECONTEXT"),
            )
            ...`
and then calling it with the optional arguments, without setting environment variables:
`kube_client = KubeClient(config_file = DEFAULT_KUBECTL_CONFIG_PATH, context = cluster)`
should do the trick for us, without breaking other things that still take their configs from env vars, right?

Reply:
yeah pretty much, allthough imo this is a bit easier to read:

```python
class KubeClient:
    def __init__(self, component: Optional[str] = None, config_file: Optional[str] = None, context: Optional[str] = None) -> None:
        if not config_file:
            config_file = os.environ.get("KUBECONFIG", KUBE_CONFIG_PATH)
        if not context:
            context = os.environ.get("KUBECONTEXT")
        kube_config.load_kube_config(
            config_file=config_file,
            context=context,
        )
        ...
```